fvm/src/kernel/default.rs

```diff
@@ -616,20 +609,26 @@ impl<C> NetworkOps for DefaultKernel<C>
 where
     C: CallManager,
 {
-    fn network_epoch(&self) -> ChainEpoch {
-        self.call_manager.context().epoch
-    }
-
-    fn network_version(&self) -> NetworkVersion {
-        self.call_manager.context().network_version
-    }
-
-    fn network_base_fee(&self) -> &TokenAmount {
-        &self.call_manager.context().base_fee
-    }
-
-    fn tipset_timestamp(&self) -> u64 {
-        self.call_manager.context().timestamp
+    fn network_context(&self) -> Result<NetworkContext> {
+        // TODO: charge gas.
+        let MachineContext {
+            epoch,
+            timestamp,
+            base_fee,
+            network: NetworkConfig {
+                network_version, ..
+            },
+            ..
+        } = self.call_manager.context();
+        Ok(NetworkContext {
+            epoch: *epoch,
+            network_version: *network_version as u32,
+            timestamp: *timestamp,
+            base_fee: base_fee
+                .try_into()
+                .or_fatal()
+                .context("base-fee exceeds u128 limit")?,
```

Comment:
I'd _prefer_ to do this conversion in the syscall itself, but then we'd need to define _two_ different `NetworkContext` types. It just isn't worth it.

Reply:
Similar caching considerations here.